### PR TITLE
checkstyle, pmd, findbugs fix and XSLT transform

### PR DESCRIPTION
**Pull request by fansu**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -512,7 +513,112 @@
...
+                        <goals>
+                            <goal>transform</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <transformationSets>
+                        <transformationSet>
+                            <dir>${main.basedir}/core/definitions</dir>
+                            <includes>html-models.tssl</includes>
```
Comment (nvsnvikram) [inline pom.xml]: Seems like <includes> should have one or more <include> files.

File: pom.xml [hunk 2]
```diff
@@ -512,7 +513,112 @@
...
+                            <includes>html-scanner.stml</includes>
+                            <stylesheet>${main.basedir}/core/stml/stml-scanner.xslt</stylesheet>
+                        </transformationSet>
+                    </transformationSets>
+                </configuration>
+                <dependencies>
+                    <dependency>
+                        <groupId>net.sf.saxon</groupId>
+                        <artifactId>saxon</artifactId>
+                        <version>8.7</version>
```
Comment (nvsnvikram) [inline pom.xml]: Is this the latest version?

Comment (fansu) [reply pom.xml]: yes, it is latest version

File: pom.xml [hunk 3]
```diff
@@ -512,7 +513,112 @@
...
+                    </execution>
+                    <execution>
+                        <id>replace-schema</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>replace</goal>
+                        </goals>
+                        <configuration>
+                            <file>${main.basedir}/core/templates/HTMLSchema.java</file>
+                            <outputFile>${main.basedir}/target/generated-sources/HTMLSchema.java</outputFile>
```
Comment (fansu) [inline pom.xml]: maintain structure with com/lafaspot/tagchowder/templates/

**Pull request merged**